src/backend/distributed/master/master_metadata_utility.c

```diff
@@ -1389,6 +1389,37 @@ EnsureFunctionOwner(Oid functionId)
 }
 
 
+/*
+ * EnsureDistributedTable error out if the given relation is not a distributed table
+ * with the given message.
+ */
+void
+EnsureDistributedTable(Oid sourceRelationId, const char *errorMessage)
+{
+	if (!IsDistributedTable(sourceRelationId))
+	{
+		ereport(ERROR, (errmsg("%s", errorMessage)));
+	}
+}
+
+
+/*
+ * IsDistributedTable returns true if the given relation is
+ * a distributed table.
+ */
+bool
+IsDistributedTable(Oid relationId)
```

Comment:
it is quite quite possible that we have such a method in our codebase, yet I couldnt find it.